vignettes/tcplfit2-vignette.Rmd

```diff
@@ -161,3 +166,159 @@ The input data for this example contains 6 signatures for one chemical in a tran
   }
 ```
 
+## Example 4: Running tcpl-like multi-concentration response data without a database connection
+The ToxCast pipeline `tcpl` is an R package that manages, curve-fits, plots, and stores ToxCast data to populate its linked MySQL database, InvitroDB. The original `tcplFit()` function within `tcpl` performed basic concentration response curve fitting. Processing with tcpl_v3 and beyond depends on `tcplfit2` to allow a wider variety of concentration-response models when using `invitrodb` in the 4.0 schema and beyond. Within this update, `tcplLite` became deprecated within `tcpl` because `tcplFit2` can be used to curve-fit data and make hitcalls independent of invitrodb, as the example below illustrates.  For additional information, please consult vignettes for `library(tcpl)` at https://CRAN.R-project.org/package=tcpl. 
+
+The input for this example comes from the ACEA_AR assay. Data from the assay component ACEA_AR_agonist_80hr was analyzed in the positive analysis fitting direction relative to DMSO as the neutral control and baseline of activity. Using a electrical impedance as a cell growth reporter, increased activity can be used to infer increased signaling at the pathway-level for the androgen receptor (as encoded by the AR gene). Given heterogeneous assay data, source data often must go through pre-processing steps to transform into a uniform data format, often like this level 0. The below table is identical to the multi-concentration level 0 data (mc0) table that would be seen in `invitrodb` and recognized by `tcpl`. Columns include:
+
+* m0id = Level 0 id
+* spid = Sample id
+* acid = Unique assay component id; unique numeric id for each assay component
+* apid = Assay plate id
+* coli = Column index (location on assay plate)
+* rowi = Row index (location on assay plate)
+* wllt = well type
+* wllq = well quality
+* conc = concentration
+* rval = raw value
+* srcf = Source file name
+* clowder_uid = clowder unique id for source files
+* git_hash = hash key for pre-processing scripts
+
+```{r example4_init, fig.height = 6, fig.width = 7, message=FALSE, warning = FALSE, echo = FALSE}
+load(file = "../data/mc0.rda")
+library(tcpl)
+library(data.table)
+dat <- mc0
+DT::datatable(head(dat[wllt=='t',]),rownames= FALSE, options = list(scrollX = T))
+```
+To run standalone `tcplfit2` fitting without the need for a MySQL database connection like `invitrodb`, the user will replicate stepping through the multiple levels of processing. A detailed explanation of processing levels can be found within `tcpl`'s Data Processing vignette.
+
+Level 1 importantly establishes the concentration index. The concentration index is simply the distinct concentrations ranked from lowest to highest, and this index can be used to calculate the baseline median absolute deviation for an assay.
+
+```{r example4_cndx, fig.height = 6, fig.width = 7}
```

Comment:
May not be necessary right now, but may be useful for the audience to have some 'in-line' comments for steps under general "step sections" (e.g. line 207 what is the action being done, those familiar with data.tables may be able to read this but those that are more of a dplyr user - like myself - this is a little more cryptic).   Just something to consider...

Reply:
This whole section will need to be reworked at some point.  I agree that data.table is cryptic to read and I am not a fan either.  I think with some more thought we could get this whole section down to a single line that defines concentration index by grouping.  However, this is the way that tcpl does it so I wanted to be sure that the process was identical.

Reply:
Makes sense, and something maybe we can plan out for the next FY or two.